Sources/MapboxNavigation/NavigationMapView+RoadNameLabeling.swift

```diff
@@ -0,0 +1,171 @@
+            } catch {
+                NSLog("Failed to add \(sourceIdentifier) with error: \(error.localizedDescription).")
+            }
+        }
+        
+        guard let mapboxStreetsSource = mapView.streetsSources().first else { return }
+        
+        let identifierNamespace = Bundle.mapboxNavigation.bundleIdentifier ?? ""
+        let roadLabelStyleLayerIdentifier = "\(identifierNamespace).roadLabels"
+        let roadLabelLayer = try? mapView.mapboxMap.style.layer(withId: roadLabelStyleLayerIdentifier) as? LineLayer
+        
+        if roadLabelLayer == nil {
+            var streetLabelLayer = LineLayer(id: roadLabelStyleLayerIdentifier)
+            streetLabelLayer.source = mapboxStreetsSource.id
+            
+            var sourceLayerIdentifier: String? {
+                let identifiers = mapView.tileSetIdentifiers(mapboxStreetsSource.id,
+                                                             sourceType: mapboxStreetsSource.type.rawValue)
+                if VectorSource.isMapboxStreets(identifiers) {
+                    return identifiers.compactMap({ VectorSource.roadLabelLayerIdentifiersByTileSetIdentifier[$0] }).first
+                }
+                
+                return nil
+            }
+            
+            streetLabelLayer.sourceLayer = sourceLayerIdentifier
+            streetLabelLayer.lineOpacity = .constant(1.0)
+            streetLabelLayer.lineWidth = .constant(20.0)
+            streetLabelLayer.lineColor = .constant(.init(.white))
+            
+            if ![DirectionsProfileIdentifier.walking, DirectionsProfileIdentifier.cycling].contains(router.routeProgress.routeOptions.profileIdentifier) {
+                // Filter out to road classes valid only for motor transport.
+                let filter = Exp(.inExpression) {
+                    "class"
+                    "motorway"
+                    "motorway_link"
+                    "trunk"
+                    "trunk_link"
+                    "primary"
+                    "primary_link"
+                    "secondary"
+                    "secondary_link"
+                    "tertiary"
+                    "tertiary_link"
+                    "street"
+                    "street_limited"
+                    "roundabout"
+                }
+                
+                streetLabelLayer.filter = filter
+            }
+            
+            do {
+                var layerPosition: MapboxMaps.LayerPosition? = nil
+                if let firstLayerIdentifier = mapView.mapboxMap.style.allLayerIdentifiers.first?.id {
+                    layerPosition = .below(firstLayerIdentifier)
+                }
+                
+                try mapView.mapboxMap.style.addLayer(streetLabelLayer, layerPosition: layerPosition)
+            } catch {
+                NSLog("Failed to add \(roadLabelStyleLayerIdentifier) with error: \(error.localizedDescription).")
```

Comment:
Replace `NSLog(_:)` with `print(_:)` I guess.

Reply:
Or even better with OSLog.

Reply:
I do not think that replacing this particular part will change anything. It'd be better to handle such logs in scope of another ticket (I think we already have at least two, which should revise logging on both iOS and CarPlay).